pkg/ruler/ruler.go

```diff
@@ -649,7 +649,7 @@ func (r *Ruler) listRuleGroupsToSyncForUsers(ctx context.Context, userIDs []stri
 	}
 
 	if len(userRings) == 0 {
-		return nil, nil
+		return map[string]rulespb.RuleGroupList{}, nil
```

Comment:
I think this fix is very tricky and hidden fix the issue. `nil` map is a valid empty map. We should handle the case where non-nil map is required at the place where it is needed.

Reply:
https://github.com/grafana/mimir/pull/5115/commits/14fe7463047f93bbde7395ed89a7aae02c1c5b78